frontend/packages/process-editor/src/components/ConfigPanel/ConfigEndEvent/CustomReceiptContent/CreateCustomReceiptForm/CreateCustomReceiptForm.tsx

```diff
@@ -46,13 +54,18 @@ export const CreateCustomReceiptForm = ({
   const updateErrors = (customReceiptForm: CustomReceiptType) => {
     const { layoutSetId, dataModelId } = customReceiptForm;
     setLayoutSetError(!layoutSetId ? t('validation_errors.required') : null);
+
     setDataModelError(
-      !dataModelId
+      !getIsDataModelValid(dataModelId)
         ? t('process_editor.configuration_panel_custom_receipt_create_data_model_error')
         : null,
     );
   };
 
+  const getIsDataModelValid = (dataModelId: string | undefined): boolean => {
+    return dataModelId && dataModelId !== NO_MODEL_KEY;
+  };
+
   const createNewCustomReceipt = (customReceipt: CustomReceiptType) => {
```

Comment:
We are still calling the mutateDataType, which from my understanding we were not suppose to do anymore? I interpreted the discussion that the dataType connection for a custom receipt layoutset was only suppose to be present in the layout-sets-file and not in appmetadata. MutateDataType mutation will also connect the task `CustomReceipt` to the specific dataType 🫣

Reply:
The only thing we need to do here is to add the dataType in the customReceiptLayoutSetConfig object 🌟